### ISSUE TITLE
Support TypeScript `"moduleResolution": "node16"`

### ISSUE THREAD

**Issue by JuanM04** [sveltejs/language-tools#1522]
### Description

In [TypeScript 4.7](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7), along the property `"module": "node16"` that enables ESM features in TypeScript, there are new values for the property `"moduleResolution"`: `"node16"` and `"nodenext"`. Both values allows importing from dependencies with [package.json exports, imports, and self-referencing](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing). Right now, this property is hard-coded to `"node"`.

### Proposed solution

The svelte-language-server shouldn't hard-code that property and instead read it from the `tsconfig.json`

**Comment by dan-peterson** [comment]
Hey @dummdidumm does this update for the language server need to be manually published to npm? Eager to try it out, but noticed NPM is listing it as 0.14.28 / updated 16 days ago.

**Comment by dummdidumm** [comment]
The fixed label means it's fixed on master but not released yet. when the issue is closed it's released. You could try it out yourself by building from sourcey the getting started instructions in the main readme if this repo should help with that

**Comment by dan-peterson** [comment]
Okay cool I tried it out but encountered an issue.

It seems to honor my module resolution settings now, and that fixes the issue importing from a module that uses the new `exports` field in `package.json`. The language server no longer reports the error: "Cannot find module x or its corresponding type declarations".

However, it seems to no longer understand svelte file imports.

The problem may be unrelated / out of the scope of svelte-language-server, or even an issue with how I built it locally, I'm not sure, but figured it was worth noting here.

---

As a base case I did a clean install of SvelteKit with the demo code and updated the tsconfig:

```json
"moduleResolution": "node16",
``` 

Now in `index.svelte` the language server reports the error:

```js
import Counter from '$lib/Counter.svelte';
```
`Cannot find module '$lib/Counter.svelte' or its corresponding type declarations.`

And dropping the alias didn't solve the issue:

```js
import Counter from '../lib/Counter.svelte';
```

`Cannot find module '../lib/Counter.svelte' or its corresponding type declarations.`


**Note that in all cases the svelte project builds and runs fine. It's just the language server that's reporting errors.**
